eventmesh-store-h2/src/main/java/org/apache/eventmesh/store/h2/schema/util/DBDataSource.java

```diff
@@ -0,0 +1,42 @@
+package org.apache.eventmesh.store.h2.schema.util;
+
+import org.apache.eventmesh.store.h2.schema.configuration.DBConfiguration;
+import org.apache.commons.dbcp2.BasicDataSource;
+
+import java.sql.Connection;
+import java.sql.SQLException;
+
+public class DBDataSource {
+    private static BasicDataSource ds = null;
+
+    private DBDataSource() {
+
+    }
+
+    public static synchronized DBDataSource createDataSource(DBConfiguration dbConfig) {
```

Comment:
The `DBDataSource` is singleton?

Reply:
Yes, DBDataSource is singleton. Basically, it performs init, start and shutdown as part of EventMeshServer init, start and shutdown process.

Reply:
But when we execute `createDataSource` method twice, we will get two `DBDataSource` instances, this is confusion, I think you would better make this class to be a real singleton.

Reply:
You are right. I pushed fixes to DBDataSource class. thx